### ISSUE TITLE
Callable[..., T] causes false positive in overload checks

### ISSUE THREAD

**Issue by ilevkivskyi** [python/mypy#6140]
This example:
```python
@overload
def deco(f: Callable[[], int]) -> Callable[[], str]: ...
@overload
def deco(f: Callable[..., int]) -> Callable[..., str]: ...
def deco(f):
    pass
```
causes
```
Overloaded function signature 2 will never be matched: signature 1's parameter type(s) are the same or broader
```
But it should be allowed (the same way we allow various last overloads with `Any` types).

cc @Michael0x2a

**Comment by Michael0x2a** [comment]
The root issue seems to be that the "will never be matched" checks call `subtypes.is_more_precise(Callable[..., int], Callable[[], int])`. This in turn calls `subtypes.is_proper_subtypes(Callable[..., int], Callable[[], int])`.

This is unfortunate because `Callable[..., int]` is indeed a subtype of `Callable[[], int]` -- but it seems we do *not* want the former to count as being "more precise" then the latter.

More generally, I think we need to modify `is_more_precise` somehow and add an extra check: if the left accepts `*args` or `**kwargs` and the right doesn't, we should return False (assuming the arguments otherwise line up).

I'm still trying to figure out if we want the former to be considered a *proper* subtype of the latter or not.
I'll run some experiments on our internal codebase + work on a fix later today.
